.azure-pipelines/impacted_area_testing/analyze_impact.py

```diff
@@ -0,0 +1,186 @@
+import ast
+import argparse
+import os
+import sys
+import pathlib
+import logging
+import json
+
+logger = logging.getLogger()
+handler = logging.StreamHandler(sys.stdout)
+formatter = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+handler.setFormatter(formatter)
+
+
+def find_python_files(directory):
+    """
+    Recursively find all python files in a given directory.
+    Args:
+        directory (str): The path to the directory.
+    Returns:
+        list: A list of file paths.
+    """
+    python_files = []
+    for root, _, files in os.walk(directory):
+        for file in files:
```

Comment:
There is also a object in python called `file`, to avoid misunderstanding, we can rename this to `script`

Reply:
Thanks. I have changed it.